### PR TITLE
[bicycle routing] Support bicycle routing in Framework::GetBestRouterand Framework::SetLastUsedRouter() to keep last used router.

### PR DESCRIPTION
**Pull request by bykoianko**
https://jira.mail.ru/browse/MAPSME-1310

@yunikkk PTAL

### CODE REVIEW

File: map/framework.cpp [hunk 1]
```diff
@@ -2441,7 +2443,12 @@ RouterType Framework::GetLastUsedRouter() const
 {
   string routerType;
   settings::Get(kRouterTypeKey, routerType);
-  return (routerType == routing::ToString(RouterType::Pedestrian) ? RouterType::Pedestrian : RouterType::Vehicle);
+
+  if (routerType == routing::ToString(RouterType::Pedestrian))
+    return  RouterType::Pedestrian;
```
Comment (ygorshenin) [inline map/framework.cpp]: clang-format, please

File: map/framework.cpp [hunk 2]
```diff
@@ -2422,6 +2422,8 @@ RouterType Framework::GetBestRouter(m2::PointD const & startPoint, m2::PointD co
   {
     if (GetLastUsedRouter() == RouterType::Pedestrian)
       return RouterType::Pedestrian;
+    if (GetLastUsedRouter() == RouterType::Bicycle)
```
Comment (ygorshenin) [inline map/framework.cpp]:
```c++
auto const lastUsedRouter = GetLastUsedRouter();
switch (lastUsedRouter)
{
  case RouterType::Pedestrian:
  case RouterType::Bicycle:
    return lastUsedRouter;
  case RouterType::Vehicle:
    // fall through
}
```

Comment (bykoianko) [inline map/framework.cpp]: Так и поступим в отдельном PR

Comment (bykoianko) [inline map/framework.cpp]: https://github.com/mapsme/omim/pull/3346

**Pull request merged**